src/main/java/de/tum/in/www1/artemis/repository/ProgrammingExerciseRepository.java

```diff
@@ -172,9 +171,14 @@
     @EntityGraph(type = LOAD, attributePaths = { "templateParticipation", "solutionParticipation", "studentParticipations" })
     Optional<ProgrammingExercise> findWithAllParticipationsById(Long exerciseId);
 
-    ProgrammingExercise findOneByTemplateParticipationId(Long templateParticipationId);
-
-    ProgrammingExercise findOneBySolutionParticipationId(Long solutionParticipationId);
+    @Query("""
+            SELECT pe FROM ProgrammingExercise pe
+            LEFT JOIN pe.studentParticipations pep
+            WHERE pep.id = :#{#participationId}
+                OR pe.templateParticipation.id = :#{#participationId}
+                OR pe.solutionParticipation.id = :#{#participationId}
+            """)
+    Optional<ProgrammingExercise> findByParticipationId(Long participationId);
```

Comment:
The guideline for SQL query params (https://docs.artemis.ase.in.tum.de/dev/guidelines/server/#proper-annotation-of-sql-query-parameters) suggest adding @ Param("participationId") before the parameter name